docs/DeveloperGuide.md

```diff
@@ -1,34 +1,182 @@
-# Developer Guide
+# Control Your Crowd - Developer Guide
 
-## Design & implementation
+* [Setting up](DeveloperGuide.md#setting-up)
+* [Design](DeveloperGuide.md#design)
+  * [Architecture](DeveloperGuide.md#architecture)
+  * [UI component](DeveloperGuide.md#ui-component)
+  * [Logic component](DeveloperGuide.md#logic-component)
+  * [Model component](DeveloperGuide.md#model-component)
+  * [Storage component](DeveloperGuide.md#storage-component)
+  * [Common classes](DeveloperGuide.md#common-classes)
+* [Implementation](DeveloperGuide.md#implementation)
+* [Documentation](DeveloperGuide.md#documentation)
+* [Testing](DeveloperGuide.md#testing)
+* [Dev Ops](DeveloperGuide.md#dev-ops)
+* [Appendix](DeveloperGuide.md#appendix)
+  * [Appendix A: Product Scope](DeveloperGuide.md#appendix-a-product-scope)
+    * [Target user profile](DeveloperGuide.md#target-user-profile)
+    * [Value proposition](DeveloperGuide.md#value-proposition)
+  * [Appendix B: User Stories](DeveloperGuide.md#appendix-b-user-stories)
+  * [Appendix C: Use Cases](DeveloperGuide.md#appendix-c-use-cases)
+  * [Appendix D: Non Functional Requirements](DeveloperGuide.md#appendix-d-non-functional-requirements)
+  * [Appendix E: Glossary](DeveloperGuide.md#appendix-e-glossary)
+  * [Appendix F: Instructions for Manual Testing](DeveloperGuide.md#appendix-f-instructions-for-manual-testing)
+    * [Launch and Shutdown](DeveloperGuide.md#launch-and-shutdown)
+    * [Clearing the list](DeveloperGuide.md#clearing-the-list)
+    * [Saving data](DeveloperGuide.md#saving-data)
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+## Setting up
 
+## Design
 
-## Product scope
-### Target user profile
+### Architecture
+
+### UI component
+
+The UI consists of `TextUi` class. It acts as a middleware between users' input
+and `Parser` class under **Logic Component**.
+
+The `UI` component,
+
+* Reads user commands using `Scanner` class.
+* Pass raw user commands into `Parser` class under **Logic Component**.
+* After the **Logic Component** has executed the function, it will then return `CommandOutput` object
+  which contains the outcome of the execution, fail or succeed.
+
+### Logic component
+
+![](images/LogicComponentStructure.png?raw=true "Structure of Logic Component")
```

Comment:
A well designed diagram with correct notations! 

For, "XYZParser = CheckInParser, FindParser, etc" it could look like assignment here. Perhaps you can make it free from any ambiguity :)

![Capture3](https://user-images.githubusercontent.com/75139323/113231774-8bb8e280-92ce-11eb-9fc4-8cd3454df0e0.PNG)